plugin/fuzz/source/dextool/plugin/fuzz/backend/fuzzvariant.d

```diff
@@ -0,0 +1,628 @@
+
+        v.accept(ns_visitor);
+
+        // fill the namespace with content from the analysis
+        root.put(ns_visitor.root);
+    }
+
+    override void visit(const(TranslationUnit) v) {
+        import std.algorithm : filter;
+        import cpptooling.analyzer.clang.type : makeLocation;
+
+        mixin(mixinNodeLog!());
+
+        LocationTag tu_loc;
+        () @trusted{ tu_loc = LocationTag(Location(v.cursor.spelling, 0, 0)); }();
+
+        v.accept(this);
+    }
+
+    void toString(Writer)(scope Writer w) @safe const {
+        import std.format : FormatSpec;
+        import std.range.primitives : put;
+
+        auto fmt = FormatSpec!char("%u");
+        fmt.writeUpToNextSpec(w);
+
+        root.toString(w, fmt);
+        put(w, "\n");
+        container.get.toString(w, FormatSpec!char("%s"));
+    }
+
+    override string toString() const {
+        import std.exception : assumeUnique;
+
+        char[] buf;
+        buf.reserve(100);
+        toString((const(char)[] s) { buf ~= s; });
+        auto trustedUnique(T)(T t) @trusted {
+            return assumeUnique(t);
+        }
+
+        return trustedUnique(buf);
+    }
+}
+
+private:
+@safe:
+
+import cpptooling.data.representation : CppRoot, CppClass, CppMethod, CppCtor,
+    CppDtor, CFunction, CppNamespace, USRType;
+import cpptooling.data.type : LocationTag, Location;
+import cpptooling.data.symbol.container : Container;
+import dsrcgen.cpp : E;
+
+enum Kind {
+    none
+}
+
+struct ImplData {
+    import cpptooling.data.type : CppMethodName;
+
+    CppRoot root;
+    alias root this;
+
+    /// Tagging of nodes in the root
+    Kind[size_t] kind;
+
+    static auto make() {
+        return ImplData(CppRoot.make);
+    }
+
+    void tag(size_t id, Kind kind_) {
+        kind[id] = kind_;
+    }
+
+    Kind lookup(size_t id) {
+        if (auto k = id in kind) {
+            return *k;
+        }
+
+        return Kind.none;
+    }
+}
+
+/*
+  Will be implemented later
+CppT rawFilter() {}
+*/
+
+
+
+/*
+  Will be implemented later
+auto translate() {}
+*/
+
+
+
+void generate(CppRoot r, Parameters params,
+        Generator.Modules modules, ref const(Container) container)
+in {
+    import std.array : empty;
+
+    assert(r.funcRange.empty);
+}
+body {
+    import std.algorithm : each, filter;
+    import std.array;
+    import std.typecons;
+    import cpptooling.data.symbol.types : USRType;
+    import cpptooling.generator.func : generateFuncImpl;
+    import cpptooling.generator.gmock : generateGmock;
+    import cpptooling.generator.includes : generateIncludes;
+    import cpptooling.data.representation;
+    import cpptooling.analyzer.type;
+
+    //generateIncludes(ctrl, params, modules.hdr);
+    foreach (incl; params.getIncludes) {
+        modules.hdr.include(cast(string) incl);
+    }
+    
+    modules.hdr.include("testingenvironment.hpp");
+    modules.hdr.include("portenvironment.hpp");
+    
```

Comment:
Should we add the source code for the runtime environment?
It would make it possible to create integration tests.
Extremely useful for regression testing.